skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurations.kt

```diff
@@ -106,18 +111,33 @@ abstract class AbstractSkaffoldRunConfiguration(
 
         XmlSerializer.serializeInto(this, element)
     }
+    override fun checkConfiguration(){
+        checkConfiguration(project)
 
-    @Throws(RuntimeConfigurationException::class)
-    override fun checkRunnerSettings(
-        runner: ProgramRunner<*>,
-        runnerSettings: RunnerSettings?,
-        configurationPerRunnerSettings: ConfigurationPerRunnerSettings?
-    ) {
-        throw RuntimeConfigurationException("Hitting function2222")
     }
 
 
-    override fun checkSettingsBeforeRun(){
-        throw RuntimeConfigurationException("Hitting function")
+    companion object {
+        var skaffoldPath: Path? = Paths.get("skaffold")
+        @JvmStatic
+        fun checkConfiguration(project:Project) {
```

Comment:
can this just live in the class itself instead of putting it in a companion object?

Reply:
Apparently Kotlin doesn't have static members of abstract classes from what I read? I wanted the function to be able to be called without creating an instance, an according to the docs that is where companion objects come in.

Reply:
makes sense.

see my other suggestion about moving this utility to `SkaffoldFileService`. This should help better structure things (since you are using this check in several contexts); as well as help make unit testing easier.